tests/scripts/check_abi_check.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+
+""" Check abi_check.py is working correctly
+"""
+
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import traceback
+import sys
+import os
+import subprocess
+import re
+
+SCRIPT_PATH = "scripts/abi_check.py"
+
+def run_abi_check(old_tag, new_tag):
+    return subprocess.check_output([SCRIPT_PATH, "-o", old_tag, "-n", new_tag],
+                                   stderr=subprocess.STDOUT).decode("utf-8")
+
+def pattern_create(lines):
+    return "^["+ "".join(["("+line+")" for line in lines]) +"]$"
```

Comment:
This doesn't do what you think. `[…]` is a _character_ set, so you're building a regex that matches a single character. Alternation (matching one of multiple patterns) is `…|…`.